src/sysupdate/sysupdate-resource.c

```diff
@@ -595,10 +600,26 @@ int resource_resolve_path(
 
                 r = get_block_device_harder_fd(fd, &d);
 
-        } else if (RESOURCE_IS_FILESYSTEM(rr->type) && root) {
+        } else if (RESOURCE_IS_FILESYSTEM(rr->type)) {
                 _cleanup_free_ char *resolved = NULL;
+                char *dollar_boot = NULL;
+                ChaseFlags chase_flags = CHASE_PREFIX_ROOT;
+
+                if (rr->path_dollar_boot) {
+                        bool unprivileged_mode = (geteuid() != 0);
+                        r = find_xbootldr_and_warn(root, NULL, unprivileged_mode, &dollar_boot, NULL, NULL);
```

Comment:
i changed find_xbootldr_and_warn() btw, please change the third argument (unprivileged_mode) to -1 now. See #27861